superset/sql_parse.py

```diff
@@ -752,11 +752,21 @@ def _extract_tables_from_sql(self) -> set[Table]:
             statements = parse(self.stripped(), dialect=self._dialect)
         except SqlglotError as ex:
             logger.warning("Unable to parse SQL (%s): %s", self._dialect, self.sql)
-            dialect = self._dialect or "generic"
+
+            if isinstance(ex, ParseError):
+                error = ex.errors[0]
+                suffix = "Check the {dialect}manual for the right syntax to use near '{highlight}' at line {line}".format(
```

Comment:
Simply doing `str(ex)` doesn't work as `SQLGlot` [includes](https://github.com/tobymao/sqlglot/blob/e5e5ba0c4921f99e063e6d1f47ffaa2181bf26e1/sqlglot/parser.py#L1264-L1273) ANSI codes which are helpful when printing in the terminal, but simply render as `[4m` and `[0m` in HTML.